### DISCUSSION TITLE
slither . doesnt run

### DISCUSSION THREAD

**Discussion by Tarunrao0** [Cyfrin/security-and-auditing-full-course-s23#93] [Q&A] [answered]
I am using WSL ubuntu and when i run `slither .`  on vs code it returns

![image](https://github.com/Cyfrin/security-and-auditing-full-course-s23/assets/122633325/d1ffbf18-56e8-4572-bf88-270564fe87e0)

**Comment by cromewar** [comment]
Hello @Tarunrao0 
based on this: https://github.com/crytic/slither-action/issues/5

Your slither installation might be interfering with another dependency or library. I highly suggest you install the tool on an isolated Python environment using `virtualenv`. 

Also, be sure you have installed the solidity solc compiler first: https://docs.soliditylang.org/en/v0.4.21/installing-solidity.html#binary-packages

**Reply by Tarunrao0** [reply, marked as answer]
Thank you sm. I was so confused when it said forge not found. The issue was somehow my foundry had uninstalled(?). Basically forge, caste and anvil werent there anymore after id installed slither. I reinstalled foundry and that solved the issue.

**Reply by EthanHuntIMF99** [reply]
@cromewar 
![image](https://github.com/Cyfrin/security-and-auditing-full-course-s23/assets/142003515/d9f4c091-55ae-4976-b94b-1bf47001c525)


I used that local env as u suggested and this is what I got. So let me just explain first of all when it was outside the src folder it was still giving the same error and now that it is inside it is saying that it can not find the files. And one more thing I thought maybe its not able to read the foundry.toml file so i tried using the direct way of accessing the files instead of remappings but it still does not work saying it can not find any files like these, can you help me out please ASAP

**Reply by alymurtazamemon** [reply]
@EthanHuntIMF99 It looks like dependencies are not installed in your project. Run `forge install` and see. 

If this don't work then your dependencies remappings are not set. For it, create a `remappings.txt` file at the root level and run `forge remappings` command on terminal. Copy paste the output in the txt file and then try to build the project using command `forge build`

**Reply by EthanHuntIMF99** [reply]
Man I did that forge install(first I also thought of that) and tried everything but still its showing the same problem

**Reply by alexxamsv** [reply]
> Man I did that forge install(first I also thought of that) and tried everything but still its showing the same problem

@EthanHuntIMF99 Hi! Have you corrected the problem? I have the same.

**Reply by EngrPips** [reply]
It seems like they re-installed `Foundry,` and everything works for them. Maybe try the same.

**Reply by alexxamsv** [reply]
> re-installed `Foundry`

@EngrPips I did it and I have the same issue again.

**Reply by EngrPips** [reply]
@alexxamsv, I will advise that you open your separate discussion and share complete details about your issue. Full details that you know will help us to help you.
